src/cdk/a11y/live-announcer.ts

```diff
@@ -28,14 +28,12 @@ export class LiveAnnouncer implements OnDestroy {
 
   constructor(
       @Optional() @Inject(LIVE_ANNOUNCER_ELEMENT_TOKEN) elementToken: any,
-      platform: Platform) {
-    // Only do anything if we're on the browser platform.
-    if (platform.isBrowser) {
-      // We inject the live element as `any` because the constructor signature cannot reference
-      // browser globals (HTMLElement) on non-browser environments, since having a class decorator
-      // causes TypeScript to preserve the constructor signature types.
-      this._liveElement = elementToken || this._createLiveElement();
-    }
+      @Inject(DOCUMENT) private _document: any) {
```

Comment:
I think you can do:
```ts
@Inject(DOCUMENT) private _document: Document|Document
```
It shouldn't preserve the type info for "computed" types; I think this will trick it will still giving type safety

Reply:
That doesn't seem to work in AoT.